conan/cli/commands/compare.py

```diff
@@ -0,0 +1,233 @@
+import json
+import subprocess
+import os
+
+from conan.errors import ConanException
+from conan.api.conan_api import ConanAPI
+from conan.api.output import ConanOutput
+from conan.cli.command import conan_command
+from conan.internal.util.files import chdir
+from conan.api.model import RecipeReference, ListPattern
+
+def output_json(msg):
+    return json.dumps({"greet": msg})
+
+
+@conan_command(group="Security", formatters={"json": output_json})
+def compare(conan_api: ConanAPI, parser, *args):
+    """
+    Command to get the diff between versions
+    """
+
+    parser.add_argument("-op", "--old-path", help="Path to the old recipe")
+    parser.add_argument("-ov", "--old-reference", help='Old reference "mylib/1.0"')
+    parser.add_argument("-or", "--old-require", help='Old reference "mylib/1.0"')
+
+    parser.add_argument("-np", "--new-path", help="Path to the new recipe")
+    parser.add_argument("-nv", "--new-reference", help="New reference")
+    parser.add_argument("-nr", "--new-require", help='New reference "mylib/1.0"')
+
+    parser.add_argument("-s", "--split_diff", action="store_true")
+    parser.add_argument("--encoding", default="utf-8", help="Encoding to read diff")
```

Comment:
Is the encoding necessary? Conan hardcodes utf-8 for all its generated files

Reply:
We're trying to figure out why git is sometimes outputting non-utf8 characters for some files (My guess is that the file it's diffing itself has a weird encoding/character sequence)